content/en/docs/concepts/scheduling-eviction/pod-eviction.md

```diff
@@ -0,0 +1,375 @@
+`--node-status-update-frequency`, which defaults to `10s`.
+
+### Node condition oscillation
+
+In some cases, nodes oscillate above and below soft eviction thresholds without
+holding for the defined grace periods. This causes the reported node condition
+to constantly switch between `true` and `false`, leading to bad eviction decisions.
+
+To protect against oscillation, you can use the `eviction-pressure-transition-period`
+flag, which controls how long the kubelet must wait before transitioning a node
+condition to a different state.
+
+## Reclaiming node level resources {#reclaim-node-resources}
+
+The kubelet tries to reclaim node-level resources before it evicts end-user pods.
+
+When a `DiskPressure` node condition is reported, the kubelet reclaims node-level
+resources based on the filesystems on the node. 
+
+### With `imagefs`
+
+If the node has a dedicated `imagefs` filesystem for container runtimes to use,
+the kubelet does the following:
+
+  * If the `nodefs` filesystem meets the eviction threshlds, the kubelet deletes
+    dead pods and containers. 
+  * If the `imagefs` filesystem meets the eviction thresholds, the kubelet
+    deletes all unused images. 
+
+### Without `imagefs`
+
+If the node only has a `nodefs` filesystem that meets eviction thresholds,
+the kubelet frees up disk space in the following order:
+
+1. Delete dead pods and containers
+1. Delete unused images
+
+## Evicting end-user pods
+
+If the kubelet's attempts to reclaim node-level resources don't bring the eviction
+signal below the threshold, the kubelet begins to evict end-user pods. 
+
+The kubelet uses the following parameters to determine pod eviction order:
+
+1. Pod resource usage relative to defined requests
+1. [Pod priority](/docs/concepts/configuration/pod-priority-preemption/)
+1. Resource consumption relative to scheduling requests
+1. Pod [QoS class](/docs/tasks/configure-pod-container/quality-service-pod/)
+
+As a result, kubelet ranks and evicts pods in the following order:
+
+1. `BestEffort` or `Burstable` pods where the usage exceeds requests. These pods
+   are evicted based on their Priority and then by how much their usage level
+   exceeds the request.
+1. `Guaranteed` pods and `Burstable` pods where the usage is less than requests
+   are evicted last, based on their Priority.
+
+`Guaranteed` pods are guaranteed only when requests and limits are specified for
+all the containers and they are equal. These pods will never be evicted because
+of another pod's resource consumption. If a system daemon (such as `kubelet`,
+`docker`, and `journald`) is consuming more resources than were reserved via 
+`system-reserved` or `kube-reserved` allocations, and the node only has
+`Guaranteed` or `Burstable` pods using less resources than requests left on it,
+then the kubelet must choose to evict one of these pods to preserve node stability
+and to limit the impact of resource starvation on other Pods. In this case, it
```

Comment:
I'm not sure regarding the `system daemon` section. 

@dashpole to keep me honest, but as far as I'm aware kubelet eviction for memory is only looking at the kubepods cgroup `statsapi.SystemContainerPods`: https://github.com/kubernetes/kubernetes/blob/fbc93bd34c51ba81aebf789da67bfa497fa9225a/pkg/kubelet/eviction/helpers.go#L672 

So I don't think memory usage outside the pod cgroup (i.e. those by system daemons) would influence kubelet memory eviction.

Reply:
Thanks! I'll wait for confirmation  and then add it.

Reply:
There are two memory eviction thresholds.

1. Sum(Pods) < Allocatable
2. All usage (including pods, system daemons) < Capacity - eviction threshold

System daemon usage isn't limited by kube-reserved in most setups, and comes into play for the second threshold.

Reply:
Leaving as-is for now.

Reply:
ah got it, thanks @dashpole for clarification. For #2 that checks is from here looks like: https://github.com/kubernetes/kubernetes/blob/fbc93bd34c51ba81aebf789da67bfa497fa9225a/pkg/kubelet/eviction/helpers.go#L666-L670 (i.e. `summary.Node.Memory`)